core-bundle/tests/Routing/ResponseContext/CoreResponseFactoryTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Routing\ResponseContext;
+
+use Contao\CoreBundle\Routing\ResponseContext\CoreResponseContextFactory;
+use Contao\CoreBundle\Routing\ResponseContext\ResponseContext;
+use Contao\CoreBundle\Routing\ResponseContext\ResponseContextAccessor;
+use Contao\CoreBundle\Routing\ResponseContext\WebpageResponseContext;
+use Contao\PageModel;
+use Contao\TestCase\ContaoTestCase;
+
+class CoreResponseFactoryTest extends ContaoTestCase
+{
+    public function testFactoryMethods(): void
+    {
+        $responseAccessor = $this->createMock(ResponseContextAccessor::class);
+        $responseAccessor
+            ->expects($this->exactly(3))
+            ->method('setResponseContext')
+        ;
+
+        $factory = new CoreResponseContextFactory($responseAccessor);
+
+        $this->assertInstanceOf(ResponseContext::class, $factory->createResponseContext());
+        $this->assertInstanceOf(WebpageResponseContext::class, $factory->createWebpageResponseContext());
```

Comment:
These tests are meaningless, because the return type is already guaranteed by the method signature.

Reply:
Only if someone calls them. If we don't, there's no coverage. And there's nothing else to test against here - I don't want to test the specific implementations. I want to test whether the factories do what they are supposed to do.

Reply:
<img width="992" alt="" src="https://user-images.githubusercontent.com/1192057/118508473-8523e480-b72f-11eb-9e53-ec080bfa3e84.png">

Reply:
IMHO the main problem isn't that this is testing against a known return type but that PHPUnit does not allow testing valid behavior of things that it cannot observe.

(see https://github.com/sebastianbergmann/phpunit-documentation/issues/171).

Reply:
You could call the factory methods without the instanceof check in the test method and add the test with `@doesNotPerformAssertions` I think.

Reply:
Changed in bd9a4a30e6e616ed086f59a501735d6499b4902a.